hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -141,6 +142,10 @@ protected AmazonS3 buildAmazonS3Client(
       // no idea what the endpoint is, so tell the SDK
       // to work it out at the cost of an extra HEAD request
       b.withForceGlobalBucketAccessEnabled(true);
+      // HADOOP-17771 force set the region so the build process doesn't halt.
+      String region = getConf().getTrimmed(AWS_REGION, AWS_S3_CENTRAL_REGION);
+      LOG.debug("Using default endpoint; setting region to {}", region);
+      b.setRegion(region);
```

Comment:
skip if region is empty?, so a config has set fs.s3a.endpoint=""; if we skip calling setRegion() there, then the connector will go to the SDK region resolution process, including picking up info from EC2 metadata